### PR TITLE
treewide: use rustPlatform.bindgenHook

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by philiptaron** [comment]
How did you find these?!

**Comment by NickCao** [comment]
> How did you find these?!

```shell
rg LIBCLANG_PATH
```

File: pkgs/os-specific/linux/scx/scx_rustscheds.nix [hunk 1]
```diff
@@ -25,6 +25,7 @@ rustPlatform.buildRustPackage {
   nativeBuildInputs = [
     pkg-config
     llvmPackages.clang
```
Comment (Aleksanaa) [inline pkgs/os-specific/linux/scx/scx_rustscheds.nix:27]: Is this still needed?

File: pkgs/by-name/ne/neothesia/package.nix [hunk 2]
```diff
@@ -35,6 +35,7 @@ rustPlatform.buildRustPackage {
     pkg-config
     llvmPackages.clang
```
Comment (Aleksanaa) [inline pkgs/by-name/ne/neothesia/package.nix:36]: Also

File: pkgs/by-name/li/libkrun/package.nix [hunk 3]
```diff
@@ -40,6 +40,7 @@ stdenv.mkDerivation (finalAttrs: {
   nativeBuildInputs = [
     llvmPackages.clang
```
Comment (Aleksanaa) [inline pkgs/by-name/li/libkrun/package.nix:41]: Also

File: pkgs/by-name/fe/fedimint/package.nix [hunk 4]
```diff
@@ -40,6 +40,7 @@ buildRustPackage rec {
     pkg-config
     clang
```
Comment (Aleksanaa) [inline pkgs/by-name/fe/fedimint/package.nix:41]: Also

File: pkgs/os-specific/linux/scx/scx_rustscheds.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/os-specific/linux/scx/scx_rustscheds.nix:27]: Let's try.

**Comment by philiptaron** [comment]
I also see the `ecc` package on that list

**Comment by NickCao** [comment]
> I also see the `ecc` package on that list; does it want similar treatment?

Nope, it's set as part of `ecc`'s wrapper.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -2345,9 +2345,7 @@ with pkgs;
 
   fdroidcl = pkgs.callPackage ../development/mobile/fdroidcl { };
 
-  fedimint = callPackage ../by-name/fe/fedimint/package.nix {
-    inherit (darwin.apple_sdk.frameworks) Security SystemConfiguration;
-  };
+  fedimint = callPackage ../by-name/fe/fedimint/package.nix { };
```
Comment (philiptaron) [inline pkgs/top-level/all-packages.nix:2348]: We can delete this line! 😀

**Comment by NickCao** [comment]
Oh no, 2086 rebuilds, I shall drop some of these.

**Comment by NickCao** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 360427`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 20 packages built:</summary>
  <ul>
    <li>caligula</li>
    <li>fedimint</li>
    <li>fedimint.devimint</li>
    <li>fedimint.fedimint</li>
    <li>fedimint.fedimintCli</li>
    <li>fedimint.gateway</li>
    <li>fedimint.gatewayCli</li>
    <li>krun</li>
    <li>krunvm</li>
    <li>libkrun</li>
    <li>libkrun-sev</li>
    <li>libkrun-sev.dev</li>
    <li>libkrun.dev</li>
    <li>neothesia</li>
    <li>nufmt</li>
    <li>scx.full</li>
    <li>scx.full.bin</li>
    <li>scx.full.dev</li>
    <li>scx.rustscheds</li>
    <li>servo</li>
  </ul>
</details>

**Comment by NickCao** [comment]
Need someone to test darwin.

**Comment by Aleksanaa** [comment]
Okay

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**